### PR TITLE
New package: nvidia-shortlived

### LINKED ISSUE #8182: Package Request: Nvidia Short-Lived Branch
Homepage: https://www.nvidia.com/object/unix.html (The second line under the "Long Lived..")

Maybe renaming the Long Lived one as "nvidia-lts" and the normal one as just "nvidia"?

This branch gives Vulkan updates and performance improvements. These are good in the name of gaming and 3d production.

### PR DESCRIPTION
**Pull request by Breavyn**
Fixes #8182.
Open to suggestions for a better name for this package. The template is based off the existing nvidia package.

### CODE REVIEW

File: srcpkgs/nvidia-shortlived/INSTALL [hunk 1]
```diff
@@ -0,0 +1,13 @@
+case "${ACTION}" in
+post)
+	# Regenerate initramfs.
```
Comment (maxice8) [inline srcpkgs/nvidia-shortlived/INSTALL]: I'd rather tabs be used here

File: srcpkgs/nvidia-shortlived/INSTALL [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	echo "Regenerating initramfs, please wait..."
+	dracut -f -q --regenerate-all
+
+	if [ "${ARCH}" = "i686" ]; then
```
Comment (maxice8) [inline srcpkgs/nvidia-shortlived/INSTALL]: why even do this ? not better to restrict archs="x86_64" ?

File: srcpkgs/nvidia-shortlived/template [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+_desc="NVIDIA drivers for linux (short lived series)"
+
+pkgname=nvidia-shortlived
+version=415.27
+revision=1
+maintainer="Colin Gillespie <colin@breavyn.com>"
+license="Proprietary NVIDIA license"
+homepage="http://www.nvidia.com"
+
+archs="i686 x86_64"
```
Comment (maxice8) [inline srcpkgs/nvidia-shortlived/template]: archs="x86_64"

File: srcpkgs/nvidia-shortlived/template [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+nopie=yes
+repository="nonfree"
+create_wrksrc=yes
+short_desc="${_desc} - Libraries and Utilities"
+conflicts="catalyst>=0 xserver-abi-video>24_1 nvidia"
+
+if [ "$XBPS_TARGET_MACHINE" = "x86_64" ]; then
+	depends="nvidia-shortlived-libs-${version}_${revision} nvidia-shortlived-gtklibs-${version}_${revision} nvidia-shortlived-dkms-${version}_${revision} pkg-config"
```
Comment (maxice8) [inline srcpkgs/nvidia-shortlived/template]: cut this at 80 columns into a newline and prefix with a whitespace, same level as depends=

File: srcpkgs/nvidia-shortlived/template [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+else
+	depends="pkg-config"
+fi
+
+build_options="glvnd"
+desc_option_glvnd="Add support for NVIDIA's GL Vendor Neutral Dispatch implementation"
+build_options_default="glvnd"
+
+if [ "$XBPS_TARGET_MACHINE" = "i686" ]; then
```
Comment (maxice8) [inline srcpkgs/nvidia-shortlived/template]: remove this part

File: srcpkgs/nvidia-shortlived/template [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+pre_install() {
+	cd ${_pkg}
+	cp nvidia_icd.json.template nvidia_icd.json
+	sed -i -e 's:__NV_VK_ICD__:libGLX_nvidia.so.0:g' nvidia_icd.json
+}
+
+do_install() {
+	cd ${_pkg}
+
+	if [ "$XBPS_TARGET_MACHINE" = "i686" ]; then
```
Comment (maxice8) [inline srcpkgs/nvidia-shortlived/template]: remove this conditional

File: srcpkgs/nvidia-shortlived/template [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+
+	vinstall libGLESv1_CM.so.1.2.0 755 usr/lib
+	ln -sf libGLESv1_CM.so.1.2.0 ${DESTDIR}/usr/lib/libGLESv1_CM.so
+	ln -sf libGLESv1_CM.so.1.2.0 ${DESTDIR}/usr/lib/libGLESv1_CM.so.1
+
+	vinstall libGLESv2.so.2.1.0 755 usr/lib
+	ln -sf libGLESv2.so.2.1.0 ${DESTDIR}/usr/lib/libGLESv2.so
+	ln -sf libGLESv2.so.2.1.0 ${DESTDIR}/usr/lib/libGLESv2.so.2
+
+	if [ "$XBPS_TARGET_MACHINE" = "x86_64" ]; then
```
Comment (maxice8) [inline srcpkgs/nvidia-shortlived/template]: remove this conditional

File: srcpkgs/nvidia-shortlived/template [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+
+	# Thread Local Storage
+	vinstall libnvidia-tls.so.${version} 755 usr/lib/
+
+	# VDPAU
+	vinstall libvdpau_nvidia.so.${version} 755 usr/lib/vdpau
+	ln -sf libvdpau_nvidia.so.${version} ${DESTDIR}/usr/lib/vdpau/libvdpau_nvidia.so.1
+
+	# misc libraries
+	if [ "$XBPS_TARGET_MACHINE" = "x86_64" ]; then
```
Comment (maxice8) [inline srcpkgs/nvidia-shortlived/template]: remove this conditional

File: srcpkgs/nvidia-shortlived/template [hunk 9]
```diff
@@ -0,0 +1,323 @@
...
+	ln -sf libnvidia-fbc.so.${version} ${DESTDIR}/usr/lib/libnvidia-fbc.so
+	ln -sf libnvidia-fbc.so.${version} ${DESTDIR}/usr/lib/libnvidia-fbc.so.1
+
+	vinstall libnvidia-fatbinaryloader.so.${version} 755 usr/lib
+	ln -sf libnvidia-fatbinaryloader.so.${version} ${DESTDIR}/usr/lib/libnvidia-fatbinaryloader.so.1
+	ln -sf libnvidia-fatbinaryloader.so.1 ${DESTDIR}/usr/lib/libnvidia-fatbinaryloader.so
+
+	# CUDA
+
+	if [ "$XBPS_TARGET_MACHINE" = "x86_64" ]; then
```
Comment (maxice8) [inline srcpkgs/nvidia-shortlived/template]: remove this conditoinal

File: srcpkgs/nvidia-shortlived/template [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+
+	vinstall libnvidia-ptxjitcompiler.so.${version} 755 usr/lib
+	ln -sf libnvidia-ptxjitcompiler.so.${version} ${DESTDIR}/usr/lib/libnvidia-ptxjitcompiler.so.1
+	ln -sf libnvidia-ptxjitcompiler.so.1 ${DESTDIR}/usr/lib/libnvidia-ptxjitcompiler.so
+
+	# helper libs for approved partners' GRID remote apps
+	vinstall libnvidia-ifr.so.${version} 755 usr/lib
+	vinstall libnvidia-fbc.so.${version} 755 usr/lib
+
+	if [ "$XBPS_TARGET_MACHINE" = "x86_64" ]; then
```
Comment (maxice8) [inline srcpkgs/nvidia-shortlived/template]: remove this conditional

File: srcpkgs/nvidia-shortlived/template [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+	fi
+
+	# License and documentation
+	vlicense LICENSE
+	vdoc README.txt README
+	vdoc NVIDIA_Changelog
+}
+
+nvidia-shortlived-gtklibs_package() {
+	short_desc="${_desc} - GTK+ libraries"
```
Comment (maxice8) [inline srcpkgs/nvidia-shortlived/template]: indent on all subpackages seem to be wrong

File: srcpkgs/nvidia-shortlived/template (hunk 4 above)
Comment (Duncaen) [reply srcpkgs/nvidia-shortlived/template]: The whole clause can go as it only supports one architecture anyways.

**Comment by maxice8** [comment]
Diff that can be applied.

```
diff --git a/srcpkgs/nvidia-shortlived/template b/srcpkgs/nvidia-shortlived/template
index af5c3753fa..0ab9a261bf 100644
--- a/srcpkgs/nvidia-shortlived/template
+++ b/srcpkgs/nvidia-shortlived/template
@@ -5,38 +5,28 @@ _desc="NVIDIA drivers for linux (short lived series)"
 pkgname=nvidia-shortlived
 version=415.27
 revision=1
+create_wrksrc=yes
+archs="x86_64"
+depends="nvidia-shortlived-libs-${version}_${revision}
+ nvidia-shortlived-gtklibs-${version}_${revision}
+ nvidia-shortlived-dkms-${version}_${revision}
+ pkg-config"
+short_desc="${_desc} - Libraries and Utilities"
 maintainer="Colin Gillespie <colin@breavyn.com>"
 license="Proprietary NVIDIA license"
 homepage="http://www.nvidia.com"
 
-archs="i686 x86_64"
+_pkg="NVIDIA-Linux-x86_64-${version}-no-compat32"
+
+distfiles="http://uk.download.nvidia.com/XFree86/Linux-x86_64/${version}/${_pkg}.run"
+checksum=0882f8e8fb3110c180edffeca23b49600bf11dd62ef72eb5db1cf9acb4fa8ccf
+
 nopie=yes
 repository="nonfree"
-create_wrksrc=yes
-short_desc="${_desc} - Libraries and Utilities"
 conflicts="catalyst>=0 xserver-abi-video>24_1 nvidia"
 
-if [ "$XBPS_TARGET_MACHINE" = "x86_64" ]; then
-	depends="nvidia-shortlived-libs-${version}_${revision} nvidia-shortlived-gtklibs-${version}_${revision} nvidia-shortlived-dkms-${version}_${revision} pkg-config"
-else
-	depends="pkg-config"
-fi
-
-build_options="glvnd"
-desc_option_glvnd="Add support for NVIDIA's GL Vendor Neutral Dispatch implementation"
-build_options_default="glvnd"
-
-if [ "$XBPS_TARGET_MACHINE" = "i686" ]; then
-	_pkg="NVIDIA-Linux-x86_64-${version}"
-	distfiles="http://uk.download.nvidia.com/XFree86/Linux-x86_64/${version}/${_pkg}.run"
-	checksum=c82b76a62035033dbfef4b9759c2b0544508e1180c8c7c1e68f35d1e6070d48b
-	subpackages="nvidia-shortlived-libs"
-else
-	_pkg="NVIDIA-Linux-x86_64-${version}-no-compat32"
-	distfiles="http://uk.download.nvidia.com/XFree86/Linux-x86_64/${version}/${_pkg}.run"
-	checksum=0882f8e8fb3110c180edffeca23b49600bf11dd62ef72eb5db1cf9acb4fa8ccf
-	subpackages="nvidia-shortlived-gtklibs nvidia-shortlived-dkms nvidia-shortlived-opencl nvidia-shortlived-libs"
-fi
+subpackages="nvidia-shortlived-gtklibs nvidia-shortlived-dkms
+ nvidia-shortlived-opencl nvidia-shortlived-libs"
 
 do_extract() {
 	install -m755 ${XBPS_SRCDISTDIR}/${pkgname}-${version}/${_pkg}.run ${wrksrc}
@@ -54,45 +44,28 @@ pre_install() {
 do_install() {
 	cd ${_pkg}
 
-	if [ "$XBPS_TARGET_MACHINE" = "i686" ]; then
-		cd 32
-	else
-		# X driver
-		vinstall nvidia_drv.so 755 usr/lib/xorg/modules/drivers
+	# X driver
+	vinstall nvidia_drv.so 755 usr/lib/xorg/modules/drivers
 
-		# GLX extension module for X
-		vinstall libglxserver_nvidia.so.${version} 755 usr/lib/xorg/modules/extensions
-		ln -sf libglxserver_nvidia.so.${version} ${DESTDIR}/usr/lib/xorg/modules/extensions/libglxserver_nvidia.so
-		ln -sf libglxserver_nvidia.so.${version} ${DESTDIR}/usr/lib/xorg/modules/extensions/libglxserver_nvidia.so.1
-	fi
+	# GLX extension module for X
+	vinstall libglxserver_nvidia.so.${version} 755 usr/lib/xorg/modules/extensions
+	ln -sf libglxserver_nvidia.so.${version} ${DESTDIR}/usr/lib/xorg/modules/extensions/libglxserver_nvidia.so
+	ln -sf libglxserver_nvidia.so.${version} ${DESTDIR}/usr/lib/xorg/modules/extensions/libglxserver_nvidia.so.1
 
 	# GLX client libs
-	if [ "${build_option_glvnd}" ]; then
-		# ----- Also provided by the libglvnd package (todo)
-		vinstall libGL.so.1.7.0 755 usr/lib
-		ln -sf libGL.so.1.7.0 ${DESTDIR}/usr/lib/libGL.so
-		ln -sf libGL.so.1.7.0 ${DESTDIR}/usr/lib/libGL.so.1
-
-		vinstall libGLX.so.0 755 usr/lib
-		ln -sf libGLX.so.0 ${DESTDIR}/usr/lib/libGLX.so
-		# --------------------------------------------------
-
-		# Required for GLVND option
-		vinstall libGLX_nvidia.so.${version} 755 usr/lib
-		ln -sf libGLX_nvidia.so.${version} ${DESTDIR}/usr/lib/libGLX_nvidia.so.0
-		ln -sf libGLX_nvidia.so.${version} ${DESTDIR}/usr/lib/libGLX_indirect.so.0
-	else
-		vinstall libGL.so.${version} 755 usr/lib
-		ln -sf libGL.so.${version} ${DESTDIR}/usr/lib/libGL.so
-		ln -sf libGL.so.${version} ${DESTDIR}/usr/lib/libGL.so.1
-
-		# Not required for non-GLVND option but recommended
-		# more info: https://devtalk.nvidia.com/default/topic/915640/multiple-glx-client-libraries-in-the-nvidia-linux-driver-installer-package/
-		vinstall libGLX_nvidia.so.${version} 755 usr/lib
-		ln -sf libGLX_nvidia.so.${version} ${DESTDIR}/usr/lib/libGLX_nvidia.so.0
-		ln -sf libGLX_nvidia.so.${version} ${DESTDIR}/usr/lib/libGLX_indirect.so.0
-
-	fi
+	# ----- Also provided by the libglvnd package (todo)
+	vinstall libGL.so.1.7.0 755 usr/lib
+	ln -sf libGL.so.1.7.0 ${DESTDIR}/usr/lib/libGL.so
+	ln -sf libGL.so.1.7.0 ${DESTDIR}/usr/lib/libGL.so.1
+
+	vinstall libGLX.so.0 755 usr/lib
+	ln -sf libGLX.so.0 ${DESTDIR}/usr/lib/libGLX.so
+	# --------------------------------------------------
+
+	# Required for GLVND option
+	vinstall libGLX_nvidia.so.${version} 755 usr/lib
+	ln -sf libGLX_nvidia.so.${version} ${DESTDIR}/usr/lib/libGLX_nvidia.so.0
+	ln -sf libGLX_nvidia.so.${version} ${DESTDIR}/usr/lib/libGLX_indirect.so.0
 
 	# OpenGL core library
 	vinstall libnvidia-glcore.so.${version} 755 usr/lib
@@ -117,13 +90,11 @@ do_install() {
 	ln -sf libGLESv2.so.2.1.0 ${DESTDIR}/usr/lib/libGLESv2.so
 	ln -sf libGLESv2.so.2.1.0 ${DESTDIR}/usr/lib/libGLESv2.so.2
 
-	if [ "$XBPS_TARGET_MACHINE" = "x86_64" ]; then
-		vinstall libnvidia-egl-wayland.so.1.1.0 755 usr/lib
-		ln -sf libnvidia-egl-wayland.so.1.1.0 ${DESTDIR}/usr/lib/libnvidia-egl-wayland.so.1
+	vinstall libnvidia-egl-wayland.so.1.1.0 755 usr/lib
+	ln -sf libnvidia-egl-wayland.so.1.1.0 ${DESTDIR}/usr/lib/libnvidia-egl-wayland.so.1
 
-		vinstall 10_nvidia.json 755 usr/share/glvnd/egl_vendor.d
-		vinstall 10_nvidia_wayland.json 755 usr/share/egl/egl_external_platform.d
-	fi
+	vinstall 10_nvidia.json 755 usr/share/glvnd/egl_vendor.d
+	vinstall 10_nvidia_wayland.json 755 usr/share/egl/egl_external_platform.d
 
 	# --------------------------------------------------
 
@@ -148,11 +119,9 @@ do_install() {
 	ln -sf libvdpau_nvidia.so.${version} ${DESTDIR}/usr/lib/vdpau/libvdpau_nvidia.so.1
 
 	# misc libraries
-	if [ "$XBPS_TARGET_MACHINE" = "x86_64" ]; then
-		vinstall libnvidia-cfg.so.${version} 755 usr/lib
-		ln -sf libnvidia-cfg.so.${version} ${DESTDIR}/usr/lib/libnvidia-cfg.so
-		ln -sf libnvidia-cfg.so.${version} ${DESTDIR}/usr/lib/libnvidia-cfg.so.1
-	fi
+	vinstall libnvidia-cfg.so.${version} 755 usr/lib
+	ln -sf libnvidia-cfg.so.${version} ${DESTDIR}/usr/lib/libnvidia-cfg.so
+	ln -sf libnvidia-cfg.so.${version} ${DESTDIR}/usr/lib/libnvidia-cfg.so.1
 
 	vinstall libnvidia-ml.so.${version} 755 usr/lib
 	ln -sf libnvidia-ml.so.${version} ${DESTDIR}/usr/lib/libnvidia-ml.so
@@ -176,12 +145,10 @@ do_install() {
 
 	# CUDA
 
-	if [ "$XBPS_TARGET_MACHINE" = "x86_64" ]; then
-		vbin nvidia-cuda-mps-control
-		vbin nvidia-cuda-mps-server
-		gzip -d nvidia-cuda-mps-control.1.gz
-		vman nvidia-cuda-mps-control.1
-	fi
+	vbin nvidia-cuda-mps-control
+	vbin nvidia-cuda-mps-server
+	gzip -d nvidia-cuda-mps-control.1.gz
+	vman nvidia-cuda-mps-control.1
 
 	vinstall libcuda.so.${version} 755 usr/lib
 	ln -sf libcuda.so.${version} ${DESTDIR}/usr/lib/libcuda.so
@@ -199,75 +166,73 @@ do_install() {
 	vinstall libnvidia-ifr.so.${version} 755 usr/lib
 	vinstall libnvidia-fbc.so.${version} 755 usr/lib
 
-	if [ "$XBPS_TARGET_MACHINE" = "x86_64" ]; then
-		# nvidia-xconfig
-		vbin nvidia-xconfig
-		gzip -d nvidia-xconfig.1.gz
-		vman nvidia-xconfig.1
-
-		# nvidia-settings
-		vbin nvidia-settings
-		gzip -d nvidia-settings.1.gz
-		vman nvidia-settings.1
-		vinstall nvidia-settings.desktop 644 usr/share/applications
-		vinstall nvidia-settings.png 644 usr/share/pixmaps
-		sed -e 's:__UTILS_PATH__:/usr/bin:' \
-			-e 's:__PIXMAP_PATH__:/usr/share/pixmaps:' \
-			-i ${DESTDIR}/usr/share/applications/nvidia-settings.desktop
-
-		vinstall libnvidia-gtk3.so.${version} 755 usr/lib
-		vinstall libnvidia-gtk2.so.${version} 755 usr/lib
-
-		# application profiles (needed by nvidia-settings)
-		vinstall nvidia-application-profiles-${version}-key-documentation 644 usr/share/nvidia
-		vinstall nvidia-application-profiles-${version}-rc 644 usr/share/nvidia
-
-		# nvidia-bug-report
-		vbin nvidia-bug-report.sh nvidia-bug-report
-		vbin nvidia-debugdump
-
-		# nvidia-smi
-		vbin nvidia-smi
-		gzip -d nvidia-smi.1.gz
-		vman nvidia-smi.1
-
-		# nvidia-modprobe
-		vinstall nvidia-modprobe 4755 usr/bin
-		gzip -d nvidia-modprobe.1.gz
-		vman nvidia-modprobe.1
-
-		# opencl pkg
-		vinstall nvidia.icd 644 etc/OpenCL/vendors
-
-		vinstall libnvidia-compiler.so.${version} 755 usr/lib
-
-		vinstall libnvidia-opencl.so.${version} 755 usr/lib
-		ln -sf libnvidia-opencl.so.${version} \
-			${DESTDIR}/usr/lib/libnvidia-opencl.so
-		ln -sf libnvidia-opencl.so.${version} \
-			${DESTDIR}/usr/lib/libnvidia-opencl.so.1
-
-		# vulkan icd
-		vinstall nvidia_icd.json 644 usr/share/vulkan/icd.d
-
-		# vulkan real-time ray tracing extensions
-		vinstall libnvidia-rtcore.so.${version} 755 usr/lib
-		vinstall libnvidia-cbl.so.${version} 755 usr/lib
-
-		# optix ray racing engine
-		vinstall libnvoptix.so.${version} 755 usr/lib
-
-		# dkms pkg
-		vmkdir usr/src/nvidia-${version}
-		vcopy "kernel/*" usr/src/nvidia-${version}
-		vcopy ${FILESDIR}/dkms.conf usr/src/nvidia-${version}
-		sed -e "s/__PKGVER/${version}/g" \
-			-e "s/__MAKEJOBS/-j$(nproc)/g" \
-			-i ${DESTDIR}/usr/src/nvidia-${version}/dkms.conf
-
-		vmkdir /usr/share/X11/xorg.conf.d/
-		vinstall nvidia-drm-outputclass.conf 644 /usr/share/X11/xorg.conf.d 30-nvidia-drm-outputclass.conf
-	fi
+	# nvidia-xconfig
+	vbin nvidia-xconfig
+	gzip -d nvidia-xconfig.1.gz
+	vman nvidia-xconfig.1
+
+	# nvidia-settings
+	vbin nvidia-settings
+	gzip -d nvidia-settings.1.gz
+	vman nvidia-settings.1
+	vinstall nvidia-settings.desktop 644 usr/share/applications
+	vinstall nvidia-settings.png 644 usr/share/pixmaps
+	sed -e 's:__UTILS_PATH__:/usr/bin:' \
+		-e 's:__PIXMAP_PATH__:/usr/share/pixmaps:' \
+		-i ${DESTDIR}/usr/share/applications/nvidia-settings.desktop
+
+	vinstall libnvidia-gtk3.so.${version} 755 usr/lib
+	vinstall libnvidia-gtk2.so.${version} 755 usr/lib
+
+	# application profiles (needed by nvidia-settings)
+	vinstall nvidia-application-profiles-${version}-key-documentation 644 usr/share/nvidia
+	vinstall nvidia-application-profiles-${version}-rc 644 usr/share/nvidia
+
+	# nvidia-bug-report
+	vbin nvidia-bug-report.sh nvidia-bug-report
+	vbin nvidia-debugdump
+
+	# nvidia-smi
+	vbin nvidia-smi
+	gzip -d nvidia-smi.1.gz
+	vman nvidia-smi.1
+
+	# nvidia-modprobe
+	vinstall nvidia-modprobe 4755 usr/bin
+	gzip -d nvidia-modprobe.1.gz
+	vman nvidia-modprobe.1
+
+	# opencl pkg
+	vinstall nvidia.icd 644 etc/OpenCL/vendors
+
+	vinstall libnvidia-compiler.so.${version} 755 usr/lib
+
+	vinstall libnvidia-opencl.so.${version} 755 usr/lib
+	ln -sf libnvidia-opencl.so.${version} \
+		${DESTDIR}/usr/lib/libnvidia-opencl.so
+	ln -sf libnvidia-opencl.so.${version} \
+		${DESTDIR}/usr/lib/libnvidia-opencl.so.1
+
+	# vulkan icd
+	vinstall nvidia_icd.json 644 usr/share/vulkan/icd.d
+
+	# vulkan real-time ray tracing extensions
+	vinstall libnvidia-rtcore.so.${version} 755 usr/lib
+	vinstall libnvidia-cbl.so.${version} 755 usr/lib
+
+	# optix ray racing engine
+	vinstall libnvoptix.so.${version} 755 usr/lib
+
+	# dkms pkg
+	vmkdir usr/src/nvidia-${version}
+	vcopy "kernel/*" usr/src/nvidia-${version}
+	vcopy ${FILESDIR}/dkms.conf usr/src/nvidia-${version}
+	vsed -e "s/__PKGVER/${version}/g" \
+		 -e "s/__MAKEJOBS/-j$(nproc)/g" \
+		 -i ${DESTDIR}/usr/src/nvidia-${version}/dkms.conf
+
+	vmkdir /usr/share/X11/xorg.conf.d/
+	vinstall nvidia-drm-outputclass.conf 644 /usr/share/X11/xorg.conf.d 30-nvidia-drm-outputclass.conf
 
 	# Blacklist nouveau
 	vmkdir usr/lib/modprobe.d
@@ -278,10 +243,6 @@ do_install() {
 	vmkdir usr/lib/dracut/dracut.conf.d
 	echo "omit_dracutmodules+=\" drm \"" > ${DESTDIR}/usr/lib/dracut/dracut.conf.d/99-nvidia.conf
 
-	if [ "$XBPS_TARGET_MACHINE" = "i686" ]; then
-		cd ..
-	fi
-
 	# License and documentation
 	vlicense LICENSE
 	vdoc README.txt README
@@ -291,17 +252,20 @@ do_install() {
 nvidia-shortlived-gtklibs_package() {
 	short_desc="${_desc} - GTK+ libraries"
 	pkg_install() {
-		vmove usr/lib/lib*gtk*.so*
+		vmove "usr/lib/lib*gtk*.so*"
 	}
 }
+
 nvidia-shortlived-libs_package() {
 	short_desc="${_desc} - common libraries"
-	provides="libEGL-${version}_${revision} libGL-${version}_${revision} libGLES-${version}_${revision}"
+	provides="libEGL-${version}_${revision}
+	 libGL-${version}_${revision} libGLES-${version}_${revision}"
 	replaces="libEGL>=0 libGL>=0 libGLES>=0"
 	pkg_install() {
 		vmove usr/lib
 	}
 }
+
 nvidia-shortlived-dkms_package() {
 	short_desc="${_desc} - DKMS kernel module"
 	dkms_modules="nvidia ${version}"
@@ -312,6 +276,7 @@ nvidia-shortlived-dkms_package() {
 		vmove usr/lib/dracut
 	}
 }
+
 nvidia-shortlived-opencl_package() {
 	short_desc="${_desc} - OpenCL implementation"
 	depends="libOpenCL"
```

**Comment by Breavyn** [comment]
Thank you for the feedback, I will definitely be more careful with formatting.

Although nvidia-shortlived{-dkms,gtklibs,opencl} are 64 bit only, the nvidia-shortlived-libs sub package is multilib. As I said this template is very similar to the existing nvidia template, perhaps creating the multilib sub package can be done a lot neater with clear intent, or even in a separate template altogether?

My personal use case where multilib libs is required; running a 32 bit program in wine that requires opengl.

How would you suggest achieving multilib nvidia-shortlived-libs?

**Comment by Breavyn** [comment]
@maxice8  I've marked your concerns as resolved as we discussed earlier. The arch conditionals have been consolidated and some minor style fixes.

**Comment by Hoshpak** [comment]
As a previous (and possibly future) use of the nvidia driver, I think it's a bad idea to add yet another version of the nvidia driver to the repository. The whole need to have an additional package only came about because somehow someone decided that the `nvidia` package had to follow the `long-lived branch`. Before that the package used to switch to the most recent version, no matter on which branch version was released because the `short-lived branch` sometimes has an older version than the other one.

From what I remember, this decision was based on the assumption that the `short-lived branch` was somehow unstable and we only should package stable software. From my understanding, that is not the case. The `long lived` branch is more like an LTS release and thus suited for distributions which want bug fixes but don't want to switch to new versions during their release cycle or at least not do it often. The `sort-lived` branch however is for people who are fine with updating more often and get the latest features in return. If we applied the same logic to other software, we would only ship ESR releases of Firefox and Libreoffice would be stuck at 6.1.5.

Now having this package would mean people using the driver would have to follow the nvidia releases and remove this package to install the regular nvidia package once the short-lived branch falls behind the long-lived one the next time. I'd rather just ship whatever is the most recent version in the `nvidia` package.

**Comment by Breavyn** [comment]
Honestly this was my preferred outcome and the initial plan was to bump nvidia to shortlived (415) and create a new package nvidia410. The shortlived branch is still stable/reliable as you said, this isn't the beta branch (418). Going from 410 to 415 shouldn't be nearly as invasive as the recent jump from 390 to 410.
If our committers are happy with this proposal I can submit a new pull request.

**Comment by Hoshpak** [comment]
superseded by #9001

**Pull request closed without merging**